x-pack/test/functional/apps/graph/feature_controls/graph_security.ts

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import expect from 'expect.js';
+import { SecurityService } from 'x-pack/test/common/services';
+import { KibanaFunctionalTestDefaultProviders } from '../../../../types/providers';
+
+// tslint:disable:no-default-export
+export default function({ getPageObjects, getService }: KibanaFunctionalTestDefaultProviders) {
+  const esArchiver = getService('esArchiver');
+  const security: SecurityService = getService('security');
+  const PageObjects = getPageObjects(['common', 'graph', 'security', 'error']);
+  const testSubjects = getService('testSubjects');
+  const appsMenu = getService('appsMenu');
+
+  describe('security', () => {
+    before(async () => {
+      esArchiver.load('empty_kibana');
+      // ensure we're logged out so we can login as the appropriate users
+      await PageObjects.security.logout();
+    });
+
+    after(async () => {
+      // logout, so the other tests don't accidentally run as the custom users we're testing below
+      await PageObjects.security.logout();
+    });
+
+    describe('global graph all privileges', () => {
+      before(async () => {
+        await security.role.create('global_graph_all_role', {
+          elasticsearch: {
+            indices: [{ names: ['logstash-*'], privileges: ['read', 'view_index_metadata'] }],
+          },
+          kibana: [
+            {
+              feature: {
+                graph: ['all'],
+              },
+              spaces: ['*'],
+            },
+          ],
+        });
+
+        await security.user.create('global_graph_all_user', {
+          password: 'global_graph_all_user-password',
+          roles: ['global_graph_all_role'],
+          full_name: 'test user',
+        });
+
+        await PageObjects.security.login(
+          'global_graph_all_user',
+          'global_graph_all_user-password',
+          {
+            expectSpaceSelector: false,
+          }
+        );
+      });
+
+      after(async () => {
+        await security.role.delete('global_graph_all_role');
+        await security.user.delete('global_graph_all_user');
+      });
+
+      it('shows graph navlink', async () => {
+        const navLinks = await appsMenu.readLinks();
+        expect(navLinks.map((link: Record<string, string>) => link.text)).to.eql([
+          'Graph',
+          'Management',
+        ]);
+      });
+
+      it('shows save button', async () => {
+        await PageObjects.common.navigateToApp('graph');
+        await testSubjects.existOrFail('graphSaveButton', 20000);
```

Comment:
Can you explain why we want a fixed timeout here and below?

Reply:
I copied this from the discover feature controls test which was hard-coding these timeouts to larger values to get CI to pass at some point, but we should be able to remove them. I'll make it so.